src/Compilers/Core/Portable/Syntax/SyntaxTrivia.cs

```diff
@@ -420,8 +419,7 @@ public Location GetLocation()
         /// </summary>
         public IEnumerable<Diagnostic> GetDiagnostics()
         {
-            // https://github.com/dotnet/roslyn/issues/40773
-            return this.SyntaxTree!.GetDiagnostics(this);
+            return this.SyntaxTree?.GetDiagnostics(this) ?? SpecializedCollections.EmptyEnumerable<Diagnostic>();
```

Comment:
>SpecializedCollections.EmptyEnumerable<Diagnostic>() [](http://example.com/codeflow?start=60&length=52)

> Matches behavior of syntax token

Looking at implementation for SyntaxToken, it looks like it might return non-empty result even when SyntaxTree is null.